### PR TITLE
Update Magento_BraintreeGraphQl module ReadMe

### PR DESCRIPTION
**Pull request by diazwatson**
### Description (*)

Updates ReadMe for the Magento_BraintreeGraphQl module.

### Fixed Issues (if relevant)

1. magento/devdocs https://github.com/magento/devdocs/issues/4451: README content for modules - work in M2 Code

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by pmclain — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @pmclain, thank you for the review.
[ENGCOM-5921](https://jira.corp.magento.com/browse/ENGCOM-5921) has been created to process this Pull Request

File: app/code/Magento/BraintreeGraphQl/README.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
-# BraintreeGraphQl
+# Magento_BraintreeGraphQl module
 
-**BraintreeGraphQl** provides type and resolver for method additional
-information.
\ No newline at end of file
+The Magento_BraintreeGraphQl module provides type and resolver for method additional information. 
+
+## Extensibility
```
Comment (kalpmehta) [inline app/code/Magento/BraintreeGraphQl/README.md:5]: @diazwatson This addition only instructs extension developers. As this is general instruction text which can be applicable to all the GraphQl modules, we may have to add this to all the GraphQl modules to maintain consistency if we start doing it. 

Alternatively, extension developers can go directly on the DevDocs site as that site is created specifically for that purpose. What are your thoughts?

**Review by kalpmehta — CHANGES_REQUESTED** [review]

File: app/code/Magento/BraintreeGraphQl/README.md (hunk 1 above)
Comment (diazwatson) [reply app/code/Magento/BraintreeGraphQl/README.md:5]: Hi @kalpmehta 
Yes you are right when you say this is a bit generic but at the same time I think it is better than the current version of the readme file which does not provide any instruction on how the module should be extended. 

Yes, all the modules in Magento2 will contain this paragraph "at least", this includes all the GraphQl modules.

Also, don't forget that this PR is part of a bigger task https://github.com/magento/devdocs/issues/4451 which aims not just to update the module's ReadMe but also bring consistency across them.

What do you think?

Comment (kalpmehta) [reply app/code/Magento/BraintreeGraphQl/README.md:5]: Great, thanks @diazwatson. I think this is a good idea. Can we add this text to all the GraphQl README as well?

Comment (diazwatson) [reply app/code/Magento/BraintreeGraphQl/README.md:5]: Sure, it is going to be added to all modules

**Review by kalpmehta — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @kalpmehta, thank you for the review.
[ENGCOM-5921](https://jira.corp.magento.com/browse/ENGCOM-5921) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable

File: app/code/Magento/BraintreeGraphQl/README.md [hunk 2]
```diff
@@ -1,4 +1,9 @@
-# BraintreeGraphQl
+# Magento_BraintreeGraphQl module
 
-**BraintreeGraphQl** provides type and resolver for method additional
-information.
\ No newline at end of file
+The Magento_BraintreeGraphQl module provides type and resolver for method additional information.
```
Comment (jeff-matthews) [inline app/code/Magento/BraintreeGraphQl/README.md]: This sentence seems incomplete. Does the module provide the type and resolver methods? What are we referring to when we say "additional information"?

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

File: app/code/Magento/BraintreeGraphQl/README.md (hunk 2 above)
Comment (diazwatson) [reply app/code/Magento/BraintreeGraphQl/README.md]: Thanks @jeff-matthews I will review

File: app/code/Magento/BraintreeGraphQl/README.md [hunk 3]
```diff
@@ -1,4 +1,9 @@
-# BraintreeGraphQl
+# Magento_BraintreeGraphQl module
 
-**BraintreeGraphQl** provides type and resolver for method additional
-information.
\ No newline at end of file
+The Magento_BraintreeGraphQl module provides type and resolver information for the GraphQl module to pass payment information data from the client to Magento.
```
Comment (jeff-matthews) [inline app/code/Magento/BraintreeGraphQl/README.md]:
```suggestion
The Magento_BraintreeGraphQl module provides type and resolver information for the GraphQL module to pass payment information data from the client to Magento.
```

**Review by jeff-matthews — CHANGES_REQUESTED** [review]

**Review by jeff-matthews — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @jeff-matthews, thank you for the review.
[ENGCOM-5921](https://jira.corp.magento.com/browse/ENGCOM-5921) has been created to process this Pull Request

**Pull request merged**